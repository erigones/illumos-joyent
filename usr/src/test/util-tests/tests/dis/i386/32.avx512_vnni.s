/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2019 Joyent, Inc.
 * Copyright 2024 Oxide Computer Company
 */

/*
 * 32-bit AVX-512 VNNI instruction disassembly.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vpdpbusd	%xmm0, %xmm1, %xmm2
	vpdpbusd	%ymm3, %ymm4, %ymm5
	vpdpbusd	%zmm6, %zmm7, %zmm0
	vpdpbusd	%xmm1, %xmm2, %xmm3{%k1}
	vpdpbusd	%xmm4, %xmm5, %xmm6{%k2}{z}
	vpdpbusd	%ymm7, %ymm0, %ymm1{%k3}
	vpdpbusd	%ymm2, %ymm3, %ymm4{%k4}{z}
	vpdpbusd	%zmm5, %zmm6, %zmm7{%k5}
	vpdpbusd	%zmm0, %zmm1, %zmm2{%k6}{z}
	vpdpbusd	0x64(%eax), %zmm3, %zmm4
	vpdpbusd	0x123456(%ebx, %ecx, 8), %zmm5, %zmm6
	vpdpbusd	(%edx){1to4}, %xmm1, %xmm2
	vpdpbusd	0x23(%edx){1to4}, %xmm1, %xmm2
	vpdpbusd	(%edx){1to8}, %ymm1, %ymm2
	vpdpbusd	0x23(%edx){1to8}, %ymm1, %ymm2
	vpdpbusd	(%edx){1to16}, %zmm1, %zmm2
	vpdpbusd	0x23(%edx){1to16}, %zmm1, %zmm2

	vpdpbusds	%xmm0, %xmm1, %xmm2
	vpdpbusds	%ymm3, %ymm4, %ymm5
	vpdpbusds	%zmm6, %zmm7, %zmm0
	vpdpbusds	%xmm1, %xmm2, %xmm3{%k1}
	vpdpbusds	%xmm4, %xmm5, %xmm6{%k2}{z}
	vpdpbusds	%ymm7, %ymm0, %ymm1{%k3}
	vpdpbusds	%ymm2, %ymm3, %ymm4{%k4}{z}
	vpdpbusds	%zmm5, %zmm6, %zmm7{%k5}
	vpdpbusds	%zmm0, %zmm1, %zmm2{%k6}{z}
	vpdpbusds	0x64(%eax), %zmm3, %zmm4
	vpdpbusds	0x123456(%ebx, %ecx, 8), %zmm5, %zmm6
	vpdpbusds	(%edx){1to4}, %xmm1, %xmm2
	vpdpbusds	0x23(%edx){1to4}, %xmm1, %xmm2
	vpdpbusds	(%edx){1to8}, %ymm1, %ymm2
	vpdpbusds	0x23(%edx){1to8}, %ymm1, %ymm2
	vpdpbusds	(%edx){1to16}, %zmm1, %zmm2
	vpdpbusds	0x23(%edx){1to16}, %zmm1, %zmm2

	vpdpwssd	%xmm0, %xmm1, %xmm2
	vpdpwssd	%ymm3, %ymm4, %ymm5
	vpdpwssd	%zmm6, %zmm7, %zmm0
	vpdpwssd	%xmm1, %xmm2, %xmm3{%k1}
	vpdpwssd	%xmm4, %xmm5, %xmm6{%k2}{z}
	vpdpwssd	%ymm7, %ymm0, %ymm1{%k3}
	vpdpwssd	%ymm2, %ymm3, %ymm4{%k4}{z}
	vpdpwssd	%zmm5, %zmm6, %zmm7{%k5}
	vpdpwssd	%zmm0, %zmm1, %zmm2{%k6}{z}
	vpdpwssd	0x64(%eax), %zmm3, %zmm4
	vpdpwssd	0x123456(%ebx, %ecx, 8), %zmm5, %zmm6
	vpdpwssd	(%edx){1to4}, %xmm1, %xmm2
	vpdpwssd	0x23(%edx){1to4}, %xmm1, %xmm2
	vpdpwssd	(%edx){1to8}, %ymm1, %ymm2
	vpdpwssd	0x23(%edx){1to8}, %ymm1, %ymm2
	vpdpwssd	(%edx){1to16}, %zmm1, %zmm2
	vpdpwssd	0x23(%edx){1to16}, %zmm1, %zmm2

	vpdpwssds	%xmm0, %xmm1, %xmm2
	vpdpwssds	%ymm3, %ymm4, %ymm5
	vpdpwssds	%zmm6, %zmm7, %zmm0
	vpdpwssds	%xmm1, %xmm2, %xmm3{%k1}
	vpdpwssds	%xmm4, %xmm5, %xmm6{%k2}{z}
	vpdpwssds	%ymm7, %ymm0, %ymm1{%k3}
	vpdpwssds	%ymm2, %ymm3, %ymm4{%k4}{z}
	vpdpwssds	%zmm5, %zmm6, %zmm7{%k5}
	vpdpwssds	%zmm0, %zmm1, %zmm2{%k6}{z}
	vpdpwssds	0x64(%eax), %zmm3, %zmm4
	vpdpwssds	0x123456(%ebx, %ecx, 8), %zmm5, %zmm6
	vpdpwssds	(%edx){1to4}, %xmm1, %xmm2
	vpdpwssds	0x23(%edx){1to4}, %xmm1, %xmm2
	vpdpwssds	(%edx){1to8}, %ymm1, %ymm2
	vpdpwssds	0x23(%edx){1to8}, %ymm1, %ymm2
	vpdpwssds	(%edx){1to16}, %zmm1, %zmm2
	vpdpwssds	0x23(%edx){1to16}, %zmm1, %zmm2
.size libdis_test, [.-libdis_test]
