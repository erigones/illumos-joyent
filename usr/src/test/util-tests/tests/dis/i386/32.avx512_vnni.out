    libdis_test:       62 f2 75 08 50 d0  vpdpbusd %xmm0,%xmm1,%xmm2
    libdis_test+0x6:   62 f2 5d 28 50 eb  vpdpbusd %ymm3,%ymm4,%ymm5
    libdis_test+0xc:   62 f2 45 48 50 c6  vpdpbusd %zmm6,%zmm7,%zmm0
    libdis_test+0x12:  62 f2 6d 09 50 d9  vpdpbusd %xmm1,%xmm2,%xmm3{%k1}
    libdis_test+0x18:  62 f2 55 8a 50 f4  vpdpbusd %xmm4,%xmm5,%xmm6{%k2}{z}
    libdis_test+0x1e:  62 f2 7d 2b 50 cf  vpdpbusd %ymm7,%ymm0,%ymm1{%k3}
    libdis_test+0x24:  62 f2 65 ac 50 e2  vpdpbusd %ymm2,%ymm3,%ymm4{%k4}{z}
    libdis_test+0x2a:  62 f2 4d 4d 50 fd  vpdpbusd %zmm5,%zmm6,%zmm7{%k5}
    libdis_test+0x30:  62 f2 75 ce 50 d0  vpdpbusd %zmm0,%zmm1,%zmm2{%k6}{z}
    libdis_test+0x36:  62 f2 65 48 50 a0  vpdpbusd 0x64(%eax),%zmm3,%zmm4
                       64 00 00 00 
    libdis_test+0x40:  62 f2 55 48 50 b4  vpdpbusd 0x123456(%ebx,%ecx,8),%zmm5,%zmm6
                       cb 56 34 12 00 
    libdis_test+0x4b:  62 f2 75 18 50 12  vpdpbusd (%edx){1to4},%xmm1,%xmm2
    libdis_test+0x51:  62 f2 75 18 50 92  vpdpbusd 0x23(%edx){1to4},%xmm1,%xmm2
                       23 00 00 00 
    libdis_test+0x5b:  62 f2 75 38 50 12  vpdpbusd (%edx){1to8},%ymm1,%ymm2
    libdis_test+0x61:  62 f2 75 38 50 92  vpdpbusd 0x23(%edx){1to8},%ymm1,%ymm2
                       23 00 00 00 
    libdis_test+0x6b:  62 f2 75 58 50 12  vpdpbusd (%edx){1to16},%zmm1,%zmm2
    libdis_test+0x71:  62 f2 75 58 50 92  vpdpbusd 0x23(%edx){1to16},%zmm1,%zmm2
                       23 00 00 00 
    libdis_test+0x7b:  62 f2 75 08 51 d0  vpdpbusds %xmm0,%xmm1,%xmm2
    libdis_test+0x81:  62 f2 5d 28 51 eb  vpdpbusds %ymm3,%ymm4,%ymm5
    libdis_test+0x87:  62 f2 45 48 51 c6  vpdpbusds %zmm6,%zmm7,%zmm0
    libdis_test+0x8d:  62 f2 6d 09 51 d9  vpdpbusds %xmm1,%xmm2,%xmm3{%k1}
    libdis_test+0x93:  62 f2 55 8a 51 f4  vpdpbusds %xmm4,%xmm5,%xmm6{%k2}{z}
    libdis_test+0x99:  62 f2 7d 2b 51 cf  vpdpbusds %ymm7,%ymm0,%ymm1{%k3}
    libdis_test+0x9f:  62 f2 65 ac 51 e2  vpdpbusds %ymm2,%ymm3,%ymm4{%k4}{z}
    libdis_test+0xa5:  62 f2 4d 4d 51 fd  vpdpbusds %zmm5,%zmm6,%zmm7{%k5}
    libdis_test+0xab:  62 f2 75 ce 51 d0  vpdpbusds %zmm0,%zmm1,%zmm2{%k6}{z}
    libdis_test+0xb1:  62 f2 65 48 51 a0  vpdpbusds 0x64(%eax),%zmm3,%zmm4
                       64 00 00 00 
    libdis_test+0xbb:  62 f2 55 48 51 b4  vpdpbusds 0x123456(%ebx,%ecx,8),%zmm5,%zmm6
                       cb 56 34 12 00 
    libdis_test+0xc6:  62 f2 75 18 51 12  vpdpbusds (%edx){1to4},%xmm1,%xmm2
    libdis_test+0xcc:  62 f2 75 18 51 92  vpdpbusds 0x23(%edx){1to4},%xmm1,%xmm2
                       23 00 00 00 
    libdis_test+0xd6:  62 f2 75 38 51 12  vpdpbusds (%edx){1to8},%ymm1,%ymm2
    libdis_test+0xdc:  62 f2 75 38 51 92  vpdpbusds 0x23(%edx){1to8},%ymm1,%ymm2
                       23 00 00 00 
    libdis_test+0xe6:  62 f2 75 58 51 12  vpdpbusds (%edx){1to16},%zmm1,%zmm2
    libdis_test+0xec:  62 f2 75 58 51 92  vpdpbusds 0x23(%edx){1to16},%zmm1,%zmm2
                       23 00 00 00 
    libdis_test+0xf6:  62 f2 75 08 52 d0  vpdpwssd %xmm0,%xmm1,%xmm2
    libdis_test+0xfc:  62 f2 5d 28 52 eb  vpdpwssd %ymm3,%ymm4,%ymm5
    libdis_test+0x102: 62 f2 45 48 52 c6  vpdpwssd %zmm6,%zmm7,%zmm0
    libdis_test+0x108: 62 f2 6d 09 52 d9  vpdpwssd %xmm1,%xmm2,%xmm3{%k1}
    libdis_test+0x10e: 62 f2 55 8a 52 f4  vpdpwssd %xmm4,%xmm5,%xmm6{%k2}{z}
    libdis_test+0x114: 62 f2 7d 2b 52 cf  vpdpwssd %ymm7,%ymm0,%ymm1{%k3}
    libdis_test+0x11a: 62 f2 65 ac 52 e2  vpdpwssd %ymm2,%ymm3,%ymm4{%k4}{z}
    libdis_test+0x120: 62 f2 4d 4d 52 fd  vpdpwssd %zmm5,%zmm6,%zmm7{%k5}
    libdis_test+0x126: 62 f2 75 ce 52 d0  vpdpwssd %zmm0,%zmm1,%zmm2{%k6}{z}
    libdis_test+0x12c: 62 f2 65 48 52 a0  vpdpwssd 0x64(%eax),%zmm3,%zmm4
                       64 00 00 00 
    libdis_test+0x136: 62 f2 55 48 52 b4  vpdpwssd 0x123456(%ebx,%ecx,8),%zmm5,%zmm6
                       cb 56 34 12 00 
    libdis_test+0x141: 62 f2 75 18 52 12  vpdpwssd (%edx){1to4},%xmm1,%xmm2
    libdis_test+0x147: 62 f2 75 18 52 92  vpdpwssd 0x23(%edx){1to4},%xmm1,%xmm2
                       23 00 00 00 
    libdis_test+0x151: 62 f2 75 38 52 12  vpdpwssd (%edx){1to8},%ymm1,%ymm2
    libdis_test+0x157: 62 f2 75 38 52 92  vpdpwssd 0x23(%edx){1to8},%ymm1,%ymm2
                       23 00 00 00 
    libdis_test+0x161: 62 f2 75 58 52 12  vpdpwssd (%edx){1to16},%zmm1,%zmm2
    libdis_test+0x167: 62 f2 75 58 52 92  vpdpwssd 0x23(%edx){1to16},%zmm1,%zmm2
                       23 00 00 00 
    libdis_test+0x171: 62 f2 75 08 53 d0  vpdpwssds %xmm0,%xmm1,%xmm2
    libdis_test+0x177: 62 f2 5d 28 53 eb  vpdpwssds %ymm3,%ymm4,%ymm5
    libdis_test+0x17d: 62 f2 45 48 53 c6  vpdpwssds %zmm6,%zmm7,%zmm0
    libdis_test+0x183: 62 f2 6d 09 53 d9  vpdpwssds %xmm1,%xmm2,%xmm3{%k1}
    libdis_test+0x189: 62 f2 55 8a 53 f4  vpdpwssds %xmm4,%xmm5,%xmm6{%k2}{z}
    libdis_test+0x18f: 62 f2 7d 2b 53 cf  vpdpwssds %ymm7,%ymm0,%ymm1{%k3}
    libdis_test+0x195: 62 f2 65 ac 53 e2  vpdpwssds %ymm2,%ymm3,%ymm4{%k4}{z}
    libdis_test+0x19b: 62 f2 4d 4d 53 fd  vpdpwssds %zmm5,%zmm6,%zmm7{%k5}
    libdis_test+0x1a1: 62 f2 75 ce 53 d0  vpdpwssds %zmm0,%zmm1,%zmm2{%k6}{z}
    libdis_test+0x1a7: 62 f2 65 48 53 a0  vpdpwssds 0x64(%eax),%zmm3,%zmm4
                       64 00 00 00 
    libdis_test+0x1b1: 62 f2 55 48 53 b4  vpdpwssds 0x123456(%ebx,%ecx,8),%zmm5,%zmm6
                       cb 56 34 12 00 
    libdis_test+0x1bc: 62 f2 75 18 53 12  vpdpwssds (%edx){1to4},%xmm1,%xmm2
    libdis_test+0x1c2: 62 f2 75 18 53 92  vpdpwssds 0x23(%edx){1to4},%xmm1,%xmm2
                       23 00 00 00 
    libdis_test+0x1cc: 62 f2 75 38 53 12  vpdpwssds (%edx){1to8},%ymm1,%ymm2
    libdis_test+0x1d2: 62 f2 75 38 53 92  vpdpwssds 0x23(%edx){1to8},%ymm1,%ymm2
                       23 00 00 00 
    libdis_test+0x1dc: 62 f2 75 58 53 12  vpdpwssds (%edx){1to16},%zmm1,%zmm2
    libdis_test+0x1e2: 62 f2 75 58 53 92  vpdpwssds 0x23(%edx){1to16},%zmm1,%zmm2
                       23 00 00 00 
